lib/ts/Arena.h

```diff
@@ -166,4 +167,34 @@ Arena::str_store(const char *str, size_t len)
   return mem;
 }
 
-#endif /* __ARENA_H__ */
+struct JemallocArena : public std::allocator<uint64_t>
```

Comment:
JemallocArena is a bad name, agreed, as Jemalloc + Arena means something very precise.

That's because Arena.cc is kinda weird name for a string-storage system .. but that's all it's used for.    In order to gain the benefit of Jemalloc and Thread-cache stuff (and to test it also) ... we need to pull out code that pre-allocates and partitions hoarded memory.

Mostly because we cannot track leaks / failures / profiling at all when memory is grabbed and returned to someone's internal cache system.